src/app/account/invitation/[inviteId]/account-form.tsx

```diff
@@ -78,24 +112,78 @@ const SetupAccountForm: FC<InviteProps & { onComplete(): void }> = ({ inviteId,
 
     return (
         <form onSubmit={form.onSubmit((values) => createAccount(values))}>
-            <Title order={3} mb="md" ta="center">
-                Enter the below information to configure your account
-            </Title>
-
             <Flex direction="column" gap="lg" w={500} mx="auto">
-                <TextInput key={form.key('firstName')} {...form.getInputProps('firstName')} label="First Name" />
-
-                <TextInput key={form.key('lastName')} {...form.getInputProps('lastName')} label="Last Name" />
-
+                <Text size="md">
+                    You&apos;ve been invited to join {orgName}. Please fill out the details below to create your
+                    account.
+                </Text>
+                <TextInput label="Email" value={email} disabled />
+
+                <Group grow gap="md">
+                    <TextInput
+                        key={form.key('firstName')}
+                        {...form.getInputProps('firstName')}
+                        label="First name"
+                        placeholder="Enter your first name"
+                        error={form.errors.firstName && <InputError error={form.errors.firstName} />}
+                    />
+
+                    <TextInput
+                        key={form.key('lastName')}
+                        {...form.getInputProps('lastName')}
+                        label="Last name"
+                        placeholder="Enter your last name"
+                        error={form.errors.lastName && <InputError error={form.errors.lastName} />}
+                    />
+                </Group>
                 <PasswordInput
-                    withAsterisk
-                    label="Password"
+                    label="Enter password"
                     key={form.key('password')}
+                    placeholder="********"
                     {...form.getInputProps('password')}
+                    error={undefined} // prevent the password input from showing an error in favor of the custom requirements below
+                />
+
+                {(() => {
+                    const lengthMet = form.values.password.length >= 8
+                    const numberMet = /[0-9]/.test(form.values.password)
+                    const upperMet = /[A-Z]/.test(form.values.password)
+                    const specialMet = /[^A-Za-z0-9]/.test(form.values.password)
+                    const allMet = lengthMet && numberMet && upperMet && specialMet
+
+                    if (!(form.values.password || form.errors.password) || allMet) return null
+
```

Comment:
likewise can the logic on reset be extracted and shared with this?  

https://github.com/safeinsights/management-app/blob/main/src/app/account/reset-password/pending-reset.tsx#L149